paket.dependencies

```diff
@@ -44,3 +44,9 @@ group Build
   nuget ILRepack
   
   github fsharp/FAKE modules/Octokit/Octokit.fsx
+  
+  nuget 0x53A.ReferenceAssemblies.net45 0.1.0
```

Comment:
So this is for net45 support, correct? because in the future this is no longer required as soon as we are fine with net461 compat?

Reply:
No, these are the reference assemblies. When we move to net461 we need to update to the correct reference assemblies for that.

The reference assemblies are the "contract assemblies" that the compiler compiles against. They are metadata only.

I used https://github.com/jaredpar/xcopy-msbuild, this basically just zipped up the content of ``C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5``

Reply:
But once we are on net461 all we need is the netstandard build, so we could drop that in theory?
I don't think this is the correct thing to do long term (though I haven't really decided yet).

I thought we will have all the reference assemblies as "official" nuget packages now?

Reply:
I haven't found _any_ reference assemblies as nuget packages.

Reply:
Otherwise I'm pretty sure the roslyn build would use these, instead of packaging this stuff up themselves.

Reply:
> But once we are on net461 all we need is the netstandard build

at the least paket.exe needs to compile against a "real" netfx version, so we still need the ref assemblies. paket.core _could_ be compiled against netstandard, but why bother?

Reply:
Hm, that's what I'm doing with FAKE 5, I don't depend on anything for compilation but the reference assemblies in the nuget packages (`ref` folder).

But I don't know if they are available for net45 as well, but I assume they are. Just look in the netstandard16 packages or the Netstandard.Library2.0 package

Reply:
> why bother?

I just thought there is a "better" way by using official infrastructure. Might be wrong ...

Reply:
What I meant is, why do 

``Paket (net461) -> Paket.Core (netstandard1.6)``

instead of

``Paket (net461) -> Paket.Core (net461)``

?

The second way surely causes less issues.

Reply:
> Just look in the netstandard16 packages or the Netstandard.Library2.0 package

I haven't looked, but I suspect these would be the reference assemblies for netstandard.

At the end of the day, paket.exe needs to target full-fx in some version, so we need the reference assemblies for full-fx.

Reply:
For example https://www.nuget.org/packages/System.Linq/ contains a reference assembly in the ref/net463 folder.

But I have no idea where they went for sdk20

Reply:
To back off a step: My understanding was that you can download the SDK and build everything - not only netcore app but also full framework applications. Is there really a requirement to install Visual Studio to get reference assemblies?

Reply:
These are all just ref assemblies for the implementation assemblies.

For example, in System.Linq: it contains neither ref nor implementation assembly for net45 (and both for net463).

We need the ref-assemblies for the assemblies of the net framework. Any nuget package would just contain a large list of ref assemblies without any implementation assemblies.

![image](https://user-images.githubusercontent.com/4236651/29734788-7b9d9ac6-89f4-11e7-9b88-88581e34933e.png)

Reply:
> My understanding was that you can download the SDK and build everything - not only netcore app but also full framework applications. Is there really a requirement to install Visual Studio to get reference assemblies?

No, but you DO need to install the targeting packs

Reply:
Yeah I get that and like that. Can you elaborate/link the RoslynTools.ReferenceAssemblies you are referring to above (I cannot find it)? And also why it doesn't work for us (just trying to figure out if we can make it work or if it is viable for other scenarios)?

Thanks for your work on this and being patient until I get it ;)

Reply:
> Can you elaborate/link the RoslynTools.ReferenceAssemblies you are referring to above (I cannot find it)?

source https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json

> And also why it doesn't work for us (just trying to figure out if we can make it work or if it is viable for other scenarios)?

It does not contain the ref-assemblies for 4.5:

![image](https://user-images.githubusercontent.com/4236651/29735124-e7f449ac-89f6-11e7-8c0f-05c30328f124.png)

Reply:
Hm, maybe I can use this?
![image](https://user-images.githubusercontent.com/4236651/29735173-2a935d7a-89f7-11e7-8764-c001f9700c1b.png)
Will try it ...

Reply:
Maybe should just upgrade to net46? I'm not sure about the portable stuff could be a subset..

Reply:
So, I did a test, and I am pretty sure we need .NETFramework\v4.5 to compile against net45 AND .NETPortable\v4.5 to ___consume___ portable assemblies, for example FsCheck.

Reply:
I have created a package that contains only these two. Reason: it is smaller.